### PR TITLE
Add script to generate authors list from git repo

### LINKED ISSUE #15: write a script that gets the list of authors
I think we need to have a file that check all the repositories for contributors and puts them together as a list in a file that is linked to in the readme rather than just linking to contributors to this repo.

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds a script to clone the git repos for qiskit-terra and
qiskit-aer and pull the authors out of the git log. It clones the git
repos to a temporary directory and then runs the necessary git log
commands to pull the author information needed. It also will handle
co-authors. The git log output parsing code is based on how the PBR
python project[1] generates authors files.

[1] https://git.openstack.org/cgit/openstack-dev/pbr/

### Details and comments

Fixes: #15

### CODE REVIEW

**Comment by mtreinish** [comment]
The biggest issue I have with this right now is there is no mail mapping, so there are a lot of duplicate authors in the list because of slightly different spelling or capitalization of names, or using different email addresses over time. This is normally fixed in git with a .mailmap file but we don't have one (likely because we rely on github to do it for us). I might be able to use the github api to do this which is going to be the next step.

**Comment by jaygambetta** [comment]
can we drop the heading sections? Can we add qiskit-aqua, and qiskit-chemistry and I think it would be also good to have a minimum for adding the author. I think a total number of contributions exceeds 50 additions or deletions. @ajavadia has some comments.

File: AUTHORS [hunk 1]
```diff
@@ -0,0 +1,102 @@
+ABIGAIL J. CROSS
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Abigail J. Cross
```

File: AUTHORS [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+ANTONIO D. CORCOLES-GONZALEZ
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Antonio D. Corcoles-Gonzales
```

File: AUTHORS [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+ANTONIO MEZZACAPO
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Antonio Mezzacapo
```

File: AUTHORS [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Abdón Rodríguez Davila
+Adenilton Silva
+Adrian Chen
+Albert Frisch
+Ali Javadi
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
```

File: AUTHORS [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+Ali Javadi-Abhari
+Andreas Fuhrer
+Andrew Cross
+Anna Phan
+Anton Karazeev
+Borja Godoy Gago
+Charles Taylor
+Chris Schnabel
+ChristopheVuillot
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Christophe Vuillot
```

File: AUTHORS [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+Enrique de la Torre
+Eugene Dumitrescu
+Fran Cabrera
+Giacomo Nannicini
+Hanhee Paik
+Hiroshi Horii
+Hitomi Takahashi
+Ikko Hamamura
+Ismael Faro Sertage
+Ismail
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Ismail Yunus Akhalwaya
```

File: AUTHORS [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+Ivan Duran
+JERRY M. CHOW
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Jerry M. Chow
```

File: AUTHORS [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+Jay Gambetta
+Jessica
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Jessica Wilson
```

File: AUTHORS [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+Jesús Pérez
+Joe Hellmers
+John A. Smolin
+Jonathan A. Wildstrom
+Jorge Carballo
+Juan Cruz-Benito (juancb)
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Juan Cruz-Benito
```

File: AUTHORS [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+Juan Gomez
+Kenso Trabing
+Kevin Krsulich
+Lee James O'Riordan
+Luciano
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
```

File: AUTHORS [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+Mathias Soeken
+Matthew Treinish
+Max Reuter
+Mingi Ryu
+Paco Martín Fernández
+Paul Nation
+Peng Liu
+Pete Taylour
+Pieter Eendebak
+Rafal
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
```

File: AUTHORS [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+Steve Wood
+Takashi Imamichi
+Thomas Alexander
+Viktor Prutyanov
+Vojtech Havlicek
+Wes Turner
+Yukio Siraichi
+Yunho Maeng
+Yunong Shi
+alfrisch
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Albert Frisch
```

File: AUTHORS [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+cclauss
+chrisculver
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Chris Culver
```

File: AUTHORS [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+ewinston
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Erick Winston
```

File: AUTHORS [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+ismael faro
+itoko
+jwildstr-ibm
+klinvill
+liupibm
+martinos
+nkanazawa
+sathayen
+stefan-woerner
+woodsp
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Steve Wood
```

File: AUTHORS [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+Vojtech Havlicek
+Wes Turner
+Yukio Siraichi
+Yunho Maeng
+Yunong Shi
+alfrisch
+cclauss
+chrisculver
+ewinston
+gadial
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Gadi Aleksandrowicz
```

File: AUTHORS [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+gitcyberian
+hitomi
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Hitomi Takahashi
```

File: AUTHORS [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+ismael faro
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Ismael Faro
```

File: AUTHORS [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+itoko
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Toshinari Itoko
```

File: AUTHORS [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+jwildstr-ibm
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
```

File: AUTHORS [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+klinvill
+liupibm
+martinos
+nkanazawa
+sathayen
+stefan-woerner
+woodsp
+yaelbh
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Yael Ben-Haim
```

File: AUTHORS [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+gitcyberian
+hitomi
+ismael faro
+itoko
+jwildstr-ibm
+klinvill
+liupibm
+martinos
+nkanazawa
+sathayen
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Ninad Sathaye
```

File: AUTHORS [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+chrisculver
+ewinston
+gadial
+gitcyberian
+hitomi
+ismael faro
+itoko
+jwildstr-ibm
+klinvill
+liupibm
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Peng Liu
```

File: AUTHORS [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+martinos
+nkanazawa
+sathayen
+stefan-woerner
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Stefen Worner
```

File: AUTHORS [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+Wes Turner
+Yukio Siraichi
+Yunho Maeng
+Yunong Shi
+alfrisch
+cclauss
+chrisculver
+ewinston
+gadial
+gitcyberian
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
```

File: AUTHORS [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+jwildstr-ibm
+klinvill
+liupibm
+martinos
+nkanazawa
+sathayen
+stefan-woerner
+woodsp
+yaelbh
+yang.luh
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Yang Luh
```

File: AUTHORS [hunk 27]
```diff
@@ -0,0 +1,102 @@
...
+cclauss
+chrisculver
+ewinston
+gadial
+gitcyberian
+hitomi
+ismael faro
+itoko
+jwildstr-ibm
+klinvill
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
```

File: AUTHORS [hunk 28]
```diff
@@ -0,0 +1,102 @@
...
+liupibm
+martinos
+nkanazawa
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
Naoki Kanazawa
```

File: AUTHORS [hunk 29]
```diff
@@ -0,0 +1,102 @@
...
+ewinston
+gadial
+gitcyberian
+hitomi
+ismael faro
+itoko
+jwildstr-ibm
+klinvill
+liupibm
+martinos
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
```

**Review by ajavadia — COMMENTED** [review]
comments in line can be just accepted

File: AUTHORS [hunk 30]
```diff
@@ -0,0 +1,95 @@
...
+Takashi Imamichi
+Thomas Alexander
+Viktor Prutyanov
+Vojtech Havlicek
+Wes Turner
+Yukio Siraichi
+Yunho Maeng
+Yunong Shi
+Albert Frisch
+cclauss
```
Comment (ajavadia) [inline AUTHORS]:
```suggestion
```

File: Qiskit.bib [hunk 31]
```diff
@@ -0,0 +1,5 @@
...
+       title = {Qiskit},
```
Comment (ajavadia) [inline Qiskit.bib]:
```suggestion
       title = {Qiskit: the Quantum Information Science Kit},
```

Comment (mtreinish) [reply Qiskit.bib]: Oops, I updated the bibtex script to do this but ran it before I save the file.

**Comment by jaygambetta** [comment]
@ajavadia what do you want to do here.

**Review by ajavadia — APPROVED** [review]

**Pull request merged**